### ISSUE TITLE
[Crash]: Villager Trade Crash on world generation

### ISSUE THREAD

**Issue by Reshy** [CorgiTaco/BYG#727]
### Minecraft Version

1.18.2

### BYG Versions

Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar

### Terrablender

TerraBlender-forge-1.18.2-1.1.0.99.jar

### Mod Loader

Forge

### Mod Loader Version

4.1.0

### What happened?

Something within BYG seems to be causing crashes in reference to Villager Trades.  Might be a mod conflict but I cannot be certain.

### crash-log

https://pastebin.com/z1zwFxAy

### latest.log

[04:43:40] [Server thread/INFO]: "E:\Personal\Archive\Games\Twitch MC\Instances\1.18.2 Forge\config\byg\trades.json5" was read.
[04:43:40] [Server thread/ERROR]: Exception caught during firing event: Multiple entries with same key: none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@25143c75, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@16ab4da7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@3f3a0e35, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@41da23ed, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@45a95837} and none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@923e350, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@66870fe7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@76529d91, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@79350b8c, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@6f1a4e99}
	Index: 5
	Listeners:
		0: NORMAL
		1: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@4436ea65
		2: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@6007408a
		3: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@790fd7fc
		4: ASM: me.thonk.croptopia.events.CroptopiaVillagerTrades@5d68c8e1 initWandering(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
		5: ASM: class potionstudios.byg.BYGForgeEventsHandler appendBYGWanderingTraderTrades(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
		6: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@2268be42
		7: ASM: com.github.alexthe666.alexsmobs.event.ServerEvents@11df4a53 onWanderingTradeSetup(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
		8: ASM: class dev.architectury.registry.level.entity.trade.forge.TradeRegistryImpl onWanderingTradeRegistering(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
		9: ASM: class vectorwing.farmersdelight.common.event.VillagerEvents onWandererTrades(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
		10: ASM: class net.mehvahdjukaar.supplementaries.setup.ModSetup registerWanderingTraderTrades(Lnet/minecraftforge/event/village/WandererTradesEvent;)V
java.lang.IllegalArgumentException: Multiple entries with same key: none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@25143c75, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@16ab4da7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@3f3a0e35, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@41da23ed, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@45a95837} and none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@923e350, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@66870fe7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@76529d91, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@79350b8c, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@6f1a4e99}
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap.conflictException(ImmutableMap.java:376)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:370)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:153)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:115)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap$Builder.buildOrThrow(ImmutableMap.java:574)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:538)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.BaseMapCodec.decode(BaseMapCodec.java:44)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.UnboundedMapCodec.lambda$decode$0(UnboundedMapCodec.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.flatMap(DataResult.java:136)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.UnboundedMapCodec.decode(UnboundedMapCodec.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.Decoder.parse(Decoder.java:18)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.FieldDecoder.decode(FieldDecoder.java:29)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapCodec$1.decode(MapCodec.java:34)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.RecordCodecBuilder$Instance$5.decode(RecordCodecBuilder.java:324)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.RecordCodecBuilder$2.decode(RecordCodecBuilder.java:107)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapDecoder.lambda$compressedDecode$0(MapDecoder.java:52)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.flatMap(DataResult.java:136)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapDecoder.compressedDecode(MapDecoder.java:52)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapCodec$MapCodecCodec.decode(MapCodec.java:91)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.util.jankson.JanksonUtil.readConfig(JanksonUtil.java:111)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.readConfig(TradesConfig.java:65)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:46)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:41)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.BYGForgeEventsHandler.appendBYGWanderingTraderTrades(BYGForgeEventsHandler.java:33)
	at net.minecraftforge.eventbus.ASMEventHandler_379_BYGForgeEventsHandler_appendBYGWanderingTraderTrades_WandererTradesEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.common.VillagerTradingManager.postWandererEvent(VillagerTradingManager.java:56)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.common.VillagerTradingManager.loadTrades(VillagerTradingManager.java:43)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.server.ServerLifecycleHooks.handleServerAboutToStart(ServerLifecycleHooks.java:82)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.client.server.IntegratedServer.m_7038_(IntegratedServer.java:63)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261)
	at java.base/java.lang.Thread.run(Thread.java:833)

[04:43:40] [Server thread/ERROR]: Exception caught during firing event: Multiple entries with same key: none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@25143c75, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@16ab4da7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@3f3a0e35, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@41da23ed, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@45a95837} and none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@923e350, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@66870fe7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@76529d91, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@79350b8c, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@6f1a4e99}
	Index: 19
	Listeners:
		0: HIGH
		1: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@b27140d
		2: ASM: class dev.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		3: NORMAL
		4: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@c1082a2
		5: ASM: class de.markusbordihn.adaptiveperformancetweakscore.server.ServerManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		6: ASM: class de.markusbordihn.adaptiveperformancetweakscore.server.ServerLoad handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		7: ASM: class de.markusbordihn.adaptiveperformancetweakscore.server.ServerLevelLoad handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		8: ASM: class de.markusbordihn.adaptiveperformancetweakscore.player.PlayerPositionManager onServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		9: ASM: class de.markusbordihn.adaptiveperformancetweakscore.entity.EntityManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		10: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@6b6db7cc
		11: ASM: me.lucko.spark.forge.ForgeSparkMod@4dff9ff0 serverInit(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		12: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@4a5f7444
		13: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@796b867e
		14: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@71e60e63
		15: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@77b30444
		16: ASM: class de.markusbordihn.adaptiveperformancetweaksspawn.spawn.SpawnManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		17: ASM: class lilypuree.hyle.EventHandler onServerStarting(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		18: ASM: class de.markusbordihn.adaptiveperformancetweaksspawn.spawn.SpawnConfigManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		19: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@46ffb3a2
		20: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@63029575
		21: ASM: class de.markusbordihn.adaptiveperformancetweaksitems.entity.ItemEntityManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		22: ASM: class de.markusbordihn.adaptiveperformancetweaksitems.entity.ExperienceOrbManager handleServerAboutToStartEvent(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		23: net.minecraftforge.eventbus.EventBus$$Lambda$2840/0x00000008009f6ae8@22d60a60
		24: ASM: top.theillusivec4.polymorph.common.CommonEventsListener@2a1b307c serverAboutToStart(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		25: ASM: team.creative.creativecore.common.config.event.ConfigEventHandler@6f70e0eb startServer(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
		26: LOWEST
		27: ASM: class com.mactso.spawnbalanceutility.Main$ForgeEvents onServerAboutToStart(Lnet/minecraftforge/event/server/ServerAboutToStartEvent;)V
java.lang.IllegalArgumentException: Multiple entries with same key: none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@25143c75, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@16ab4da7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@3f3a0e35, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@41da23ed, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@45a95837} and none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@923e350, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@66870fe7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@76529d91, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@79350b8c, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@6f1a4e99}
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap.conflictException(ImmutableMap.java:376)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:370)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:153)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:115)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap$Builder.buildOrThrow(ImmutableMap.java:574)
	at MC-BOOTSTRAP/com.google.common@31.0.1-jre/com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:538)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.BaseMapCodec.decode(BaseMapCodec.java:44)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.UnboundedMapCodec.lambda$decode$0(UnboundedMapCodec.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.flatMap(DataResult.java:136)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.UnboundedMapCodec.decode(UnboundedMapCodec.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.Decoder.parse(Decoder.java:18)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.FieldDecoder.decode(FieldDecoder.java:29)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapCodec$1.decode(MapCodec.java:34)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.RecordCodecBuilder$Instance$5.decode(RecordCodecBuilder.java:324)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.codecs.RecordCodecBuilder$2.decode(RecordCodecBuilder.java:107)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapDecoder.lambda$compressedDecode$0(MapDecoder.java:52)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.DataResult.flatMap(DataResult.java:136)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapDecoder.compressedDecode(MapDecoder.java:52)
	at MC-BOOTSTRAP/datafixerupper@4.1.27/com.mojang.serialization.MapCodec$MapCodecCodec.decode(MapCodec.java:91)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.util.jankson.JanksonUtil.readConfig(JanksonUtil.java:111)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.readConfig(TradesConfig.java:65)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:46)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:41)
	at TRANSFORMER/byg@1.4-RC-10/potionstudios.byg.BYGForgeEventsHandler.appendBYGWanderingTraderTrades(BYGForgeEventsHandler.java:33)
	at net.minecraftforge.eventbus.ASMEventHandler_379_BYGForgeEventsHandler_appendBYGWanderingTraderTrades_WandererTradesEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.common.VillagerTradingManager.postWandererEvent(VillagerTradingManager.java:56)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.common.VillagerTradingManager.loadTrades(VillagerTradingManager.java:43)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at TRANSFORMER/forge@40.1.0/net.minecraftforge.server.ServerLifecycleHooks.handleServerAboutToStart(ServerLifecycleHooks.java:82)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.client.server.IntegratedServer.m_7038_(IntegratedServer.java:63)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261)
	at java.base/java.lang.Thread.run(Thread.java:833)

[04:43:40] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.IllegalArgumentException: Multiple entries with same key: none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@25143c75, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@16ab4da7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@3f3a0e35, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@41da23ed, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@45a95837} and none={2=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@923e350, 4=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@66870fe7, 1=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@76529d91, 3=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@79350b8c, 5=>[Lnet.minecraft.world.entity.npc.VillagerTrades$ItemListing;@6f1a4e99}
	at com.google.common.collect.ImmutableMap.conflictException(ImmutableMap.java:376) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:370) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:153) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:115) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.google.common.collect.ImmutableMap$Builder.buildOrThrow(ImmutableMap.java:574) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:538) ~[guava-31.0.1-jre.jar%2328!/:?]
	at com.mojang.serialization.codecs.BaseMapCodec.decode(BaseMapCodec.java:44) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.codecs.UnboundedMapCodec.lambda$decode$0(UnboundedMapCodec.java:38) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.DataResult.flatMap(DataResult.java:136) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.codecs.UnboundedMapCodec.decode(UnboundedMapCodec.java:38) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.Decoder.parse(Decoder.java:18) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.codecs.FieldDecoder.decode(FieldDecoder.java:29) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.MapCodec$1.decode(MapCodec.java:34) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder$Instance$5.decode(RecordCodecBuilder.java:324) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder$2.decode(RecordCodecBuilder.java:107) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.MapDecoder.lambda$compressedDecode$0(MapDecoder.java:52) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.DataResult.lambda$flatMap$10(DataResult.java:138) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.DataResult.flatMap(DataResult.java:136) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.MapDecoder.compressedDecode(MapDecoder.java:52) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at com.mojang.serialization.MapCodec$MapCodecCodec.decode(MapCodec.java:91) ~[datafixerupper-4.1.27.jar%2333!/:?]
	at potionstudios.byg.util.jankson.JanksonUtil.readConfig(JanksonUtil.java:111) ~[Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar%23201!/:1.4-RC-10]
	at potionstudios.byg.common.entity.npc.TradesConfig.readConfig(TradesConfig.java:65) ~[Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar%23201!/:1.4-RC-10]
	at potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:46) ~[Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar%23201!/:1.4-RC-10]
	at potionstudios.byg.common.entity.npc.TradesConfig.getConfig(TradesConfig.java:41) ~[Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar%23201!/:1.4-RC-10]
	at potionstudios.byg.BYGForgeEventsHandler.appendBYGWanderingTraderTrades(BYGForgeEventsHandler.java:33) ~[Oh_The_Biomes_You'll_Go-forge-1.18.2-1.4-RC-10-all.jar%23201!/:1.4-RC-10]
	at net.minecraftforge.eventbus.ASMEventHandler_379_BYGForgeEventsHandler_appendBYGWanderingTraderTrades_WandererTradesEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.common.VillagerTradingManager.postWandererEvent(VillagerTradingManager.java:56) ~[forge-1.18.2-40.1.0-universal.jar%23278!/:?]
	at net.minecraftforge.common.VillagerTradingManager.loadTrades(VillagerTradingManager.java:43) ~[forge-1.18.2-40.1.0-universal.jar%23278!/:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.server.ServerLifecycleHooks.handleServerAboutToStart(ServerLifecycleHooks.java:82) ~[forge-1.18.2-40.1.0-universal.jar%23278!/:?]
	at net.minecraft.client.server.IntegratedServer.m_7038_(IntegratedServer.java:63) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
[04:43:41] [Server thread/FATAL]: Preparing crash report with UUID 54581126-73bb-464c-9465-c88f80737c36
[04:43:41] [Server thread/ERROR]: This crash report has been saved to: E:\Personal\Archive\Games\Twitch MC\Instances\1.18.2 Forge\crash-reports\crash-2022-04-24_04.43.41-server.txt
[04:43:41] [Server thread/INFO]: Stopping server
[04:43:41] [Server thread/INFO]: Saving players
[04:43:41] [Server thread/INFO]: Saving worlds
[04:43:41] [Server thread/ERROR]: Exception stopping the server
java.lang.NullPointerException: Cannot invoke "net.minecraft.server.level.ServerLevel.m_6857_()" because "serverlevel2" is null
	at net.minecraft.server.MinecraftServer.m_129885_(MinecraftServer.java:544) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.server.MinecraftServer.m_7041_(MinecraftServer.java:611) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.client.server.IntegratedServer.m_7041_(IntegratedServer.java:174) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:717) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[client-1.18.2-20220404.173914-srg.jar%23274!/:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]

**Comment by Darkfiend009** [comment]
I can confirm this crash.

Minecraft: 1.18.2
Forge: 40.1.19
BYG: 1.18.2-1.4-RC-11

Crash Report: https://pastebin.com/ShB6CZHz
latest.log: https://gist.github.com/Darkfiend009/068dde695b7bea64ac772df87f6e05f9

**Comment by CorgiTaco** [comment]
@Reshy @Darkfiend009 Reset your villager trade configs, if the issue persists reopen/comment on this issue

**Comment by Darkfiend009** [comment]
It is safe to say that there is something wrong with the trade config.

I just updated to BYG 1.18.2-1.4-RC-13 and ran the command that pops up to update the configs. The game closed as it should and i relaunched without editing any configs. I received the crash listed above related to the trade config.

I deleted the trade config and relaunched and it's working again.

It's worth noting that when i reported the issue prior, i also had not edited the trade config.
